share/doc/revng/user-manual/python-scripting.md

```diff
@@ -0,0 +1,128 @@
+# Python scripting
+
+This page describes the rev.ng Python scripting capabilities.
+
+## Creating a project
+
+There two project classes to use rev.ng from Python: `CLIProject`, which spawns `revng` as a subprocess for each command, and `DaemonProject`, which interacts with `revng daemon` over the network using the GraphQL API.
+Apart from this difference, they offer the same programming interface.
+
+First, import the project of choice:
+
+```python
+>>> from revng.project import CLIProject     # for cli
+>>> from revng.project import DaemonProject  # for daemon
+```
+
+Then you must instantiate the `project` class
+
+```python
+>>> project = CLIProject()
+```
+
+You can provide a path to the resume directory (similar to `revng [command] --resume`).
+This useful if you want to persist changes for loading the project again in the future:
+
+```python
+>>> import os
+>>> resume = "path/to/resume/dir"
+>>> os.makedirs(resume)  # The directory must exists before using it
+>>> project = CLIProject(resume)
+```
+
+After that, you can to import a binary and run the *initial auto analyses*, we have a method that does just that:
+
+```python
+>>> from revng.support import get_calc_binary
+>>> binary = get_calc_binary()
+>>> project.import_and_analyze(binary)
+```
+
+## Producing artifacts
+
+You can also get the `artifacts`:
+
+```{python nodiff=2,6,9,20}
+# Get artifact through the `project`
+>>> project.get_artifact("disassemble")
+
+# Pass functions that you want to get artifacts from, the second argument is a
+# `list` so you can pass multiple functions
+>>> project.get_artifact("disassemble", [project.model.Functions[0]])
+
+# Get artifact through a `function`, this is equal to above
+>>> project.model.Functions[0].get_artifact("disassemble")
```

Comment:
Move this above the previous snippet, I expect this to be the main use case.

Reply:
Done